### ISSUE TITLE
Constant aliasing in RBI files causes type checking failures

### ISSUE THREAD

**Issue by egiurleo** [Shopify/tapioca#1594]
I recently ran into an issue when generating RBI files for the `eventmachine` and `em-synchrony` gems:

- `eventmachine` aliases the `EventMachine` constant as `EM` ([lib/eventmachine.rb:1600](https://github.com/eventmachine/eventmachine/blob/master/lib/eventmachine.rb#L1600))
- `em-synchrony` aliases `EM::S` to `EventMachine::Synchrony` ([lib/em-synchrony.rb:160](https://github.com/igrigorik/em-synchrony/blob/master/lib/em-synchrony.rb#L160))
- When both of these lines are written to RBI files, the following error is raised on the `eventmachine` RBI:

```
Cannot initialize the class or module `EM` by constant assignment[4022](https://srb.help/4022)
em-synchrony@1.0.6.rbi(13, 1): Previously defined as a class or module here
eventmachine@1.2.7.rbi(51, 1):
Sorbet does not allow treating constant assignments as class or module definitions,
even if the initializer computes a `Module` object at runtime. See the docs for more.
```

Basically, Sorbet seems to think that the `EM` constant is being defined in the `em-synchrony` RBI, and then the `eventmachine` RBI is trying to reassign it.

Is there a way we can have Tapioca generate valid RBI for a case like this?

## Reproduction steps
I've created a failing test on a branch to demonstrate this error:

1. Checkout my branch: `git fetch && get checkout emily/alias-issue`
2. Run the failing test: `bin/test spec/tapioca/cli/gem_spec.rb:2044`

cc: @morriar
